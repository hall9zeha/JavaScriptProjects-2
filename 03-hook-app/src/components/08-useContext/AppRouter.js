import React from 'react'
import { BrowserRouter as  Router, Routes, Route } from 'react-router-dom'
import { AboutScreen } from './AboutScreen'
import { HomeScreen } from './HomeScreen'
import { LoginScreen } from './LoginScreen'
import { NavBar } from './NavBar'

export const AppRouter = () => {
  return (
    <Router>
        <div>
            <NavBar/>
            {/* El Routes de react-router es como un switch tradicional(Switch en react-router-dom está descontinuado), dependiendo de qué condición ejecutará tal o cual
            En este caso una dirección a un componente dentro de la ruta */}
            <Routes>
            {/* exact le dice a la ruta que debe buscar el path con todos sus caracteres y no por la primera coincidencia */}
                <Route exact path='/' Component={HomeScreen}/> 
                <Route exact path='/login' Component={LoginScreen}/>
                <Route exact path='/about' Component={AboutScreen}/>
                {/* Ponemos una ruta por defecto si no existiera ninguna de las anteriores */}
                <Route path='*' Component={HomeScreen}/> 
            </Routes>
        </div>
    </Router>
  )
}
