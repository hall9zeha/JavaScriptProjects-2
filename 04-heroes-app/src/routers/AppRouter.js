
import React from 'react'
import { BrowserRouter, Route, Routes } from 'react-router-dom'

import { LoginScreen } from '../components/login/LoginScreen'

import { DashboardRoutes } from './DashboardRoutes'

export const AppRouter = () => {
  return (
    <BrowserRouter>
        {/* Mostraremos la pantalla de login pero sin el navbar */}
        <Routes>
            <Route path='/login' element={<LoginScreen/>}/>
            <Route path='/*' element={<DashboardRoutes/>}/>
        </Routes>
    </BrowserRouter>
  )
}
