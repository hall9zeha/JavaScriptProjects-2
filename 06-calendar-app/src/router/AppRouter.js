import React from 'react'
import {BrowserRouter , Navigate, Route, Router, Routes} from 'react-router-dom'
import { LoginScreen } from '../auth/LoginScreen'

import { CalendarScreen } from '../calendar/CalendarScreen'

export const AppRouter = () => {
  return (
    <BrowserRouter>
        <Routes>
             <Route 
                Component={LoginScreen}
                exact path='/login'/>
            <Route
                Component={CalendarScreen}
                exact path='/calendar'/>
            <Route 
                path='*' element={<Navigate to = "/calendar"/>}
            />
           </Routes>
    </BrowserRouter>
  )
}
