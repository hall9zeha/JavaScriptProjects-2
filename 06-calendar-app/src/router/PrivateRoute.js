import React from "react"

import {Navigate, Route} from 'react-router-dom'
import PropTypes from 'prop-types'


export const PrivateRoute = ({children,isAuthenticated}) => {

  return isAuthenticated ? <>{children}</>  : <Navigate to="/login"/>
}
