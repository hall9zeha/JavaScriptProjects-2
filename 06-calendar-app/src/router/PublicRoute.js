import React from "react"

import {Navigate, Route} from 'react-router-dom'
import PropTypes from 'prop-types'


export const PublicRoute = ({children,isAuthenticated}) => {

  return isAuthenticated ?  <Navigate to="/"/> :<>{children}</> 
}

