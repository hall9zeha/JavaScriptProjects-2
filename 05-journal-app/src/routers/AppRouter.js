import React, { useEffect, useState } from 'react'
import {firebase} from '../firebase/firebaseConfig'
/*
Ya que react router 5 es todavía muy utilizado este proyecto implementará dicha versión
para aprender el manejor de las rutas con la misma.
Al momento de realizar este proyecto la versión más reciente era 6.22
*/
import{
    BrowserRouter as Router,
    Switch,
    Route,
    Redirect
} from "react-router-dom";
import { PrivateRoute } from './PrivateRoute';
import { AuthRouter } from './AuthRouter';
import { JournalScreen } from '../components/journal/JournalScreen';
import { useDispatch } from 'react-redux';
import { login } from '../actions/auth';
import { PublicRoute } from './PublicRoute';
import { loadNotes } from '../helpers/loadNotes';
import { setNotes, startLoadingNotes } from '../actions/notes';


export const AppRouter = () => {

    const dispatch = useDispatch();
    const [cheking, setCheking] = useState(true);
    const [isLogged, setIsLogged] = useState(false);

    // Observador que mostrara los estados de autenticación de firebase cuando estos cambien (login or logout)
    // nos devolverá el objeto correspondiente (user credentials or null)
    useEffect(()=>{
        firebase.auth().onAuthStateChanged(async (user)=>{
            // Si el usuario no es nulo
            if(user?.uid){
                dispatch(login(user.uid,user.displayName));
                setIsLogged(true)
                dispatch(startLoadingNotes(user.uid));  

            }else{
                setIsLogged(false)
            }
            // Si deveulve datos de usuario correctamente la espera se detendrá 
            setCheking(false);   
        })
    }, [dispatch,setCheking])//[] Dependencias vacías para que se ejecute una sola vez si se vuelve a renderizar algún componente o
    // ya que dispatch y setChecking no van a cambiar tampoco se ejecutará nuevamente nustro observable, podemos incluirlos dentro [.]
    
    //Para mostrar las vistas principales esperamos a la comprobación de si estamos autenticados o no
    if(cheking){
        return (
            <h1>Espere...</h1>
        )
    }
  return (
    <Router>
        <div>
            <Switch>
                <PublicRoute
                    path="/auth"
                    component={AuthRouter}
                    isAuthenticated={isLogged}
                />
                <PrivateRoute
                    isAuthenticated={isLogged}
                    exact
                    path="/"
                    component={JournalScreen}
                />
                <Redirect to="/auth/login"/>
            </Switch>
        </div>
       
    </Router>
  )
}
