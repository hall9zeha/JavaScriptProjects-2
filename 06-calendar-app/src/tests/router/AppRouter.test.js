import React from 'react'
import {Provider} from 'react-redux';
import renderer ,{act}from 'react-test-renderer';


import configureStore from 'redux-mock-store'
import { AppRouter } from '../../router/AppRouter';


const thunk = require('redux-thunk').thunk;


//***************************/
const middlewares = [thunk]
const mockStore = configureStore(middlewares)

//Para saber con que argumentos se hizo el dispatch simulamos uno
//store.dispatch = jest.fn()


describe('Test in <AppRouter/> component', () => { 

    test('should show Prease wait...', () => { 

        const initState ={
            auth:{
                checking:true
            }
        }
        let store =  mockStore(initState);

        const wrapper = renderer.create(
            <Provider store={store}>
                <AppRouter/>
            </Provider>
        )
        expect(wrapper).toMatchSnapshot();

    })
    test('should show public route', () => { 
        const initState ={
            auth:{
                checking:false                
            }
        }
        const store =  mockStore(initState);

        const wrapper = renderer.create(
            <Provider store={store}>
                <AppRouter/>
            </Provider>
        )
        expect(wrapper.toJSON()).toMatchSnapshot();
     })

     test('should show private route', () => { 
        //TODO arreglar error: la librería react-big-calendar en la versión utilizada actualmente 
        //da el siguiete error TypeError: Cannot use 'in' operator to search for 'window' in null
        
        //Si quitamos el componente <Calendar/> de CalendarScreen la lógica de la ruta privada funciona correctamente
        const initState ={
            auth:{
                checking:false,
                uid:'1234',
                name:'Barry'
            },
            calendar:{
                events:[{title:'Test event', user:'Barry'}]
            },
            ui:{
                modalOpen:false
            }
        }
        
        const store =  mockStore(initState);
        // const wrapper = renderer.create(
        //     <Provider store={store}>
        //             <AppRouter/>
        //     </Provider>
          
        //   )
        // expect(wrapper.toJSON()).toMatchSnapshot();
     })
 })