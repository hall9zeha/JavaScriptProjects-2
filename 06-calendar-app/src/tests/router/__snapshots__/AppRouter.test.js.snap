// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <AppRouter/> component should show Prease wait... 1`] = `
<h5>
  Prease wait...
</h5>
`;

exports[`Test in <AppRouter/> component should show public route 1`] = `null`;
